rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -162,17 +164,30 @@ private void initializeClusterManager(int clusterType)
                         log.warn("Could not get nodeName", e);
                     }
 
-                    if (clusterType == HADOOP2_CLUSTER_MANAGER.ordinal()) {
-                        clusterManager = new Hadoop2ClusterManager();
-                        clusterManager.initialize(conf);
-                        nodes = clusterManager.getNodes();
-                        splitSize = clusterManager.getSplitSize();
-                    }
-                    else if (clusterType == TEST_CLUSTER_MANAGER.ordinal()) {
+                    if (clusterType == TEST_CLUSTER_MANAGER.ordinal()) {
                         nodes = new ArrayList<>();
                         nodes.add(nodeName);
                         splitSize = 64 * 1024 * 1024;
                     }
+                    else {
+                        if (clusterType == HADOOP2_CLUSTER_MANAGER.ordinal()) {
+                            clusterManager = new Hadoop2ClusterManager();
+                        }
+
+                        else if (clusterType == PRESTO_CLUSTER_MANAGER.ordinal()) {
+                            try {
+                                nodeName = InetAddress.getLocalHost().getHostAddress();
```

Comment:
this is local host address while earlier, we are storing canonical hostname.  In Presto, we need to store the host address while in hadoop and test, we need the hostname